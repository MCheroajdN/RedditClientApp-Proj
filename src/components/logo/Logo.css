.logo img {
	border-radius: 50%;
	width: 40px;
	height: 40px;
	padding: 0.3rem;
}
.logo span {
	font-family: 'Roboto', serif;
	padding-left: 2px;
	font-size: 1.5rem;
	font-weight: bold;
	color: red;
}
.logo span small {
	color: rgb(190, 66, 66);
}
.logo:hover {
	text-decoration: none;
}

@media (max-width: 1060px) {
	.logo span {
		font-size: 0.8rem;
	}
}

@media (max-width: 1035px) {
	.logo span {
		display: none;
	}
	.logo img {
		margin-right: 1rem;
	}
}
