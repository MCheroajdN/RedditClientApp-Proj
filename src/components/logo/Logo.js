import React from 'react';
import { NavLink } from 'react-router-dom';
import { showElement } from '../../utils/reusableCode';
import './Logo.css';
import logo from './logo.png';

const Logo = () => {
	const handleClick = () => {
		showElement('.popular-posts');
	};
	return (
		<NavLink className='logo' to='/' onClick={handleClick}>
			<img src={logo} alt='Logo' />
			<span>
				Reddit<small>Client</small>
				
			</span>
		</NavLink>
	);
};

export default Logo;
