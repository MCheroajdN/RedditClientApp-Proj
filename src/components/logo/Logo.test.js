import { shallow } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Logo from './Logo';

describe('Logo component', () => {
	it('should render RedditClient', () => {
		const wrapper = shallow(<Logo />);
		const logoName = 'RedditClient';
		const logoText = wrapper.find('.logo span').text();
		expect(logoText).toEqual(logoName);
	});
	it('should have logo.png image', () => {
		const expectedImage = 'logo.png';
		const wrapper = shallow(<Logo />);
		const img = wrapper.find('.logo img');

		expect(img.prop('src')).toBe(expectedImage);
	});
	it('matches the snapshot', () => {
		const tree = renderer.create(
			<Router>
				<Logo />
			</Router>
		);
		expect(tree.toJSON()).toMatchSnapshot();
	});
});
