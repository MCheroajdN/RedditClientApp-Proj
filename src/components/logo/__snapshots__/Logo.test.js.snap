// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo component has logo.png 1`] = `
<a
  aria-current="page"
  className="logo active"
  href="/"
  onClick={[Function]}
  style={Object {}}
>
  <img
    alt="Reddit Logo"
    src="logo.png"
  />
  <span>
    Reddit
    <small>
      Client
    </small>
  </span>
</a>
`;

exports[`Logo component matches the snapshot 1`] = `
<a
  aria-current="page"
  className="logo active"
  href="/"
  onClick={[Function]}
  style={Object {}}
>
  <img
    alt="Reddit Logo"
    src="logo.png"
  />
  <span>
    Simple
    <small>
      REDDIT
    </small>
  </span>
</a>
`;
